packages/appsync-emulator-serverless/serverCore.js

```diff
@@ -95,6 +95,33 @@ class SubscriptionServer {
         continue;
       }
 
+      let shouldPublish = true;
```

Comment:
I am still working through the logic but this would be better implemented IMO as a function that returns true/false and is consumed here. Most likely can avoid the mutable state too.

Reply:
This occurred to me as well.  Will implement.